hondana/collections.py

```diff
@@ -61,13 +62,47 @@
     "LegacyMappingCollection",
 )
 
+T = TypeVar("T")
+
 
 class _Collection:
     def __init__(self) -> None:
         self.type: Literal["collection"] = "collection"
 
 
-class MangaCollection(_Collection):
+class BaseCollection(ABC, Generic[T]):
+    """
+    The base class for all collections. This class serves to make it easier to create functions that process
+    arbitrary collections without needing to set up ``isinstance()`` checks.
+
+    Attributes
+    -----------
+    total: :class:`int`
+        The total possible results with this query could return.
+    offset: :class:`int`
+        The offset used in this query.
+    limit: :class:`int`
+        The limit used in this query.
+    """
+
+    total: int
+    offset: int
+    limit: int
+
```

Comment:
```suggestion

    type: Literal["collection"]
```

This would allow us to totally remove `_Collection` and simply use the ABC created, since `_Collection` was only a sentinel type to host that attribute for subclasses.

If you agree you can also remove `_Collection` above.

Reply:
I was going to ask why the `_Collection` type exists. It doesn't seem to be used anywhere nor is it documented. I figured it was for old backwards compatibility so I left it alone.

Reply:
The real reason I didn't merge these two is become most but not all collection classes inherit the `_Collection` type.

Reply:
`_Collection` exists as it is the parent class of all collection types, merely to set the `.type` attribute, as well as be available for any potential methods or attributes needed in future. This can now be handled in the ABC so feel free to remove it, or I will get it later.

Reply:
I already closed my IDE for the project, so if you could get it that would be fine. If `.type` isn't being used right now it should be fine to outright delete it, but since it is a **non-protected** attribute removing it now might become a breaking change.